cli/util/import_map.rs

```diff
@@ -59,23 +63,49 @@ impl<'a> ImportMapUnfurler<'a> {
     })?;
     let mut text_changes = Vec::new();
     let module_info = DefaultModuleAnalyzer::module_info(&parsed_source);
-    let mut analyze_specifier =
-      |specifier: &str, range: &deno_graph::PositionRange| {
+    let analyze_specifier =
+      |specifier: &str,
+       range: &deno_graph::PositionRange,
+       text_changes: &mut Vec<deno_ast::TextChange>| {
         let resolved = self.import_map.resolve(specifier, url);
         if let Ok(resolved) = resolved {
-          let new_text = if resolved.scheme() == "file" {
-            format!("./{}", url.make_relative(&resolved).unwrap())
-          } else {
-            resolved.to_string()
-          };
           text_changes.push(deno_ast::TextChange {
             range: to_range(&parsed_source, range),
-            new_text,
+            new_text: make_relative_to(url, &resolved),
           });
         }
       };
     for dep in &module_info.dependencies {
-      analyze_specifier(&dep.specifier, &dep.specifier_range);
+      match dep {
+        DependencyDescriptor::Static(dep) => {
+          analyze_specifier(
+            &dep.specifier,
+            &dep.specifier_range,
+            &mut text_changes,
+          );
+        }
+        DependencyDescriptor::Dynamic(dep) => {
+          let success = try_unfurl_dynamic_dep(
+            self.import_map,
+            url,
+            &parsed_source,
+            dep,
+            &mut text_changes,
+          );
+
+          if !success {
+            log::warn!(
+              "{} Dynamic import was not analyzable and won't use the import map once published.\n    at {}",
+              crate::colors::yellow("Warning"),
+              format_range_with_colors(&deno_graph::Range {
+                specifier: url.clone(),
+                start: dep.range.start.clone(),
+                end: dep.range.end.clone(),
+              })
+            );
+          }
```

Comment:
Could you add a test that exercised this warning?

Reply:
I wanted to, but we don't have any integration tests for publishing at the moment.